@AH-64D Apache Longbow/Addons/fza_ah64_controls/scripting/functions/core/fn_coreGetObjectsLookedAt.sqf

```diff
@@ -31,27 +31,29 @@ Author:
 #include "\fza_ah64_controls\headers\script_common.hpp"
 params ["_heli"];
 
-#define COCKPIT_CONTROL(pilot_mempoint, gunner_mempoint, system, system_name, control, sensitivity, control_name) [pilot_mempoint, gunner_mempoint, #system, #control, sensitivity, control_name]
-#define COCKPIT_CONTROL_SEP ,
+private _controls = _heli getvariable "fza_ah64_objectsDataArray";
 private _horizontalPos = _heli getVariable "fza_ah64_freeCursorHpos";
 private _verticalpos = _heli getVariable "fza_ah64_freeCursorVpos";
-private _controls =  [ //Schema of Pilot memory point, Gunner memory point, system, control, sensitivity, description]
-#include "\fza_ah64_controls\headers\controls.h"
-];
-_controls = _controls apply {
-    _x params ["_pilotPos", "_gunnerPos", "_systemName", "_eventName", "_sensitivity", "_description"];
+
+_controls select {
+    _x params ["_pilotPos", "_gunnerPos", "_systemName", "_eventName", "_sensitivity", "_description", "_movable"];
     private _point = [];
-    if(driver _heli == player && _pilotPos != "") then {
-        _point = _heli modelToWorldVisual (_heli selectionposition _pilotPos);
+    if(driver _heli == player) then {
+        if (_pilotPos isEqualTo "") then {continue;};
+        if (_movable) exitwith {
+            _point = _heli modelToWorldVisual (_heli selectionposition _pilotPos);
+        };
+        _point = _heli modelToWorldVisual _pilotPos;
     };
-    if(gunner _heli == player && _gunnerPos != "") then {
-        _point = _heli modelToWorldVisual (_heli selectionposition _gunnerPos);
+    if(gunner _heli == player) then {
+        if (_gunnerPos isEqualTo "") then {continue;};
+        if (_movable) exitwith {
+            _point = _heli modelToWorldVisual (_heli selectionposition _gunnerPos);
+        };
+        _point = _heli modelToWorldVisual _gunnerPos;
     };
     _point =  (worldToScreen _point);
-    _distance = if (_point isEqualTo []) then {1000} else {_point distance [_horizontalPos, _verticalpos]};
-    [_pilotPos, _gunnerPos, _systemName, _eventName, _sensitivity, _description, _distance];
-};
-_controls select {
-    _x params ["", "", "", "", "_sensitivity", "", "_distance"];
-    _distance < _sensitivity;
+    if (_point isEqualTo []) then {continue};
+    _distance = _point distance [_horizontalPos, _verticalpos];
+    (_distance <= _sensitivity)
```

Comment:
The reason why the `select` and `apply` are separate is because later we need the distance to do the `BIS_fnc_sortBy` - as far as I can tell that is missing here?

Reply:
This function still returns an array of mpd points within the sensitivity distance to be sorted by BIS_fnc_sortBy for interaction and clicker hints

Reply:
but i think i see what you mean that I'm no longer passing along the distance param

Reply:
Fixed